docs/frontend.md

```diff
@@ -0,0 +1,13 @@
+# Frontend
+
+```{eval-rst}
+.. currentmodule:: scipy_doctest.frontend
```

Comment:
the division between frontend.py and impl.py is definitely an implementation detail, is there a way to avoid dumping it to readers?